src/__tests__/union.test.ts

```diff
@@ -76,6 +76,59 @@ describe('union', () => {
         });
       });
 
+      describe('when none of the volumes are readable', () => {
+        it('throws error when calling a read method', () => {
+          const vol1 = Volume.fromJSON({ '/foo': 'bar1' });
+          const vol2 = Volume.fromJSON({ '/foo': 'bar2' });
+          const ufs = new Union();
+          ufs.use(vol1 as any, { readable: false }).use(vol2 as any, { readable: false });
+
+          expect(() => ufs.readFileSync('/foo')).toThrowError();
+        });
+      });
+
+      describe('when none of the volumes are writable', () => {
+        it('throws error when calling a write method', () => {
+          const vol1 = Volume.fromJSON({ '/foo': 'bar' });
+          const vol2 = Volume.fromJSON({ '/foo': 'bar' });
+          const ufs = new Union();
+          ufs.use(vol1 as any, { writable: false }).use(vol2 as any, { writable: false });
+
+          expect(() => ufs.writeFileSync('/foo', 'bar')).toThrowError();
+        });
+      });
+
+      describe('readable/writable', () => {
+        it('writes to the last vol added', () => {
+          const vol1 = Volume.fromJSON({});
+          const vol2 = Volume.fromJSON({});
+          const ufs = new Union();
+          ufs.use(vol1 as any).use(vol2 as any);
+          ufs.writeFileSync('/foo', 'bar');
+          expect(vol2.readFileSync('/foo', 'utf8')).toEqual('bar');
+        });
+
+        it('writes to the latest added writable vol', () => {
+          const vol1 = Volume.fromJSON({});
+          const vol2 = Volume.fromJSON({});
+          const ufs = new Union();
+          ufs.use(vol1 as any).use(vol2 as any, { readable: false });
+          ufs.writeFileSync('/foo', 'bar');
+
+          expect(vol2.readFileSync('/foo', 'utf8')).toEqual('bar');
+        });
+
+        it('reads from the latest added readable vol', () => {
+          const vol1 = Volume.fromJSON({'/foo': 'bar'});
+          const vol2 = Volume.fromJSON({});
+          const ufs = new Union();
+          ufs.use(vol1 as any).use(vol2 as any, { readable: false });
```

Comment:
I think these tests might be a bit mixed up.

This test looks like the equivalent to this to [this `async` test](https://github.com/streamich/unionfs/pull/441/files#diff-721fbb0cf0df35f9a41d38cc81fbf323R326), but has `readable: false` instead of `writable: false`.

Would you be able to review these, and ensure that they're correct?
It would really help if you could arrange them so that they share the same title, implementation, and order across the three sets, just so that it's easier to compare.

---------

In saying that, I think either the above is true, or we've missing tests for something.

Ultimately we should have a test for both reading & writing; I'd expect to see at least a few tests along these lines:

```
it('reads from the latest added readable vol', () => {
  const vol1 = Volume.fromJSON({'/foo': 'bar'});
  const vol2 = Volume.fromJSON({'/foo': 'baz'});
  const ufs = new Union();
  ufs.use(vol1 as any).use(vol2 as any, { readable: false });

  expect(ufs.readFileSync('/foo', 'utf8')).toEqual('bar');
  expect(vol1.readFileSync('/foo', 'utf8')).toEqual('bar');
});
```

Reply:
ok updated